### PR TITLE
core: Add properties to Operation

### PR DESCRIPTION
**Pull request by math-fehr**
This PR adds properties to operations.
Properties are inherent attributes that change the semantics of operations, and cannot be discarded.
In practice, most attribute definitions should become properties definition, which will be added in IRDL in a future PR.
Properties are a dictionary of attributes, just like the attribute dictionary, and a property may have the same name as an attribute.

### CODE REVIEW

File: xdsl/dialects/builtin.py [hunk 1]
```diff
@@ -1174,14 +1174,21 @@ class UnregisteredOpWithName(UnregisteredOp):
             @classmethod
             def create(
                 cls,
+                *,
```
Comment (superlopuh) [inline xdsl/dialects/builtin.py:1177]: this seems like a great change to separate into its own PR, would make the changes here a little more focused

**Review by PapyChacal — APPROVED** [review]
Thanks!
